### PR TITLE
Added many entity related methods

### PR DESCRIPTION
**Pull request by DaDamRival**
Added `ENTITY:getBodygroup`
Added `ENTITY:getBodygroups`
Added `ENTITY:lookupBodygroup`
Added `ENTITY:getBodygroupName`
Added `ENTITY:getSkinCount`
Added `ENTITY:isEffectActive`

Added `HOLO:addEffects`
Added `HOLO:removeEffects`
Made `HOLO:playAnimation` shared

Added `PLAYER:playGesture`
Added `PLAYER:resetGesture`

Added enum `EF` for use with the 3 effect methods
Added enum `GESTURE_SLOT` for use with the 2 gesture methods

### CODE REVIEW

File: lua/starfall/libs_sh/entities.lua [hunk 1]
```diff
@@ -1026,4 +1072,15 @@ function ents_methods:getCreationTime()
 	return ent:GetCreationTime()
 end
 
+--- Applies a engine effect to the hologram
```
Comment (thegrb93) [inline lua/starfall/libs_sh/entities.lua]: Applies should be Checks if

File: lua/starfall/libs_sh/holograms.lua [hunk 2]
```diff
@@ -509,4 +488,79 @@ function hologram_methods:setModel(model)
...
+	rate = rate or 1
+
+	holo.AutomaticFrameAdvance = animation~=-1
+
+	holo:ResetSequence(animation)
+	holo:SetCycle(frame)
+	holo:SetPlaybackRate(rate)
+end
+
+--- Applies a engine effect to the hologram
```
Comment (thegrb93) [inline lua/starfall/libs_sh/holograms.lua]: an

File: lua/starfall/libs_sh/players.lua [hunk 3]
```diff
@@ -397,6 +397,43 @@ if CLIENT then
 	function player_methods:voiceVolume()
 		getply(self):VoiceVolume()
 	end
+	
+	--- Animates a hologram
```
Comment (thegrb93) [inline lua/starfall/libs_sh/players.lua]: Animates a player

File: lua/starfall/libs_sh/players.lua [hunk 4]
```diff
@@ -397,6 +397,43 @@ if CLIENT then
...
+	-- @client
+	-- @param animation sequence string or act number. https://wiki.facepunch.com/gmod/Enums/ACT
+	-- @param loop Should the gesture loop
+	-- @param slot The gesture slot to use. GESTURE_SLOT table values
+	-- @param weight The weight of the gesture
+	function player_methods:playGesture(animation, loop, slot, weight)
+		slot = slot or GESTURE_SLOT_CUSTOM
+		checkluatype(slot, TYPE_NUMBER)
```
Comment (thegrb93) [inline lua/starfall/libs_sh/players.lua]: if slot==nil then slot = GESTURE_SLOT_CUSTOM else checkluatype(slot, TYPE_NUMBER) end

Comment (thegrb93) [reply lua/starfall/libs_sh/players.lua]: The documentation should also specify that these parameters are optional.

File: lua/starfall/libs_sh/entities.lua (hunk 1 above)
Comment (DaDamRival) [reply lua/starfall/libs_sh/entities.lua]: Shouldn't it be applied to the entity since it is for all entities and not just holograms?

Comment (thegrb93) [reply lua/starfall/libs_sh/entities.lua]: Ya entity, not hologram

**Pull request merged**